lib/puppet/type.rb

```diff
@@ -644,11 +645,13 @@ def [](name)
     name = name.intern
     fail("Invalid parameter #{name}(#{name.inspect})") unless self.class.validattr?(name)
 
-    if name == :name && nv = name_var
+    nv = name_var
+    if name == :name && nv
```

Comment:
Previously we never called `name_var` if `name == :name`, which is a common. Is calling `name_var` do anything fancy/complicated?